app/javascript/components/concept-exercise-graph/Exercise.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { ExerciseCompleteIcon } from './ExerciseCompleteIcon'
+
+import { Exercise as IExercise, ExerciseState } from './exercise-types'
+
+import './Exercise.css'
+
+export const Exercise = ({
```

Comment:
Rather than thinking of this as an `Exercise`, should we think of this as a `Concept`?

Reply:
Maybe a `ConceptExercise`? I don't think I've been thinking of them as concepts directly since some exercises may unlock more than one concept sometimes.

Reply:
> I don't think I've been thinking of them as concepts directly since some exercises may unlock more than one concept sometimes.

But it's each Concept that's getting its own node, rather than each exercise. So if one exercise taught ["strings", "booleans"] there would be two nodes still. Thus I think they're actually concepts.

Reply:
Oh. Then I think there are several assumptions I have made here. I have assumed that the nodes were exercises (like they were for the dashboard graph), not plain concepts, so there might be several things going on here that need revising.

Reply:
Yeah - sorry. It's only really today since writing the concept code that I differentiated this in my head too.,

I actually suspect not much need changing. I think the result will be the same as what's here, but probably changing L11 (3 below this) to `unlocks` and then renaming `[eE]xercise` to `[cConcept]` gets us there. Then if there are changes, it's probably in #112. My instinct is there won't be too much, though.

If you want to discuss this to get clarity, we can jump on a call tomorrow or Monday? 🙂 Sorry again for not catching it earlier!

Reply:
Good to talk about it, to summarize our conversation:

The config.json represents an exercise-exercise graph, which is a subgraph of the concept-concept graph.  This requires changes on both front- and back-end. A broad overview of the required changes:

- frontend
  - rename <Exercise /> to <Concept /> and so on...
- backend
  - when creating the graph and iterating over the exercises, flat_map the concepts to become the nodes